### PR TITLE
Improve hint when provided array for destructuring has fewer elements than expected

### LINKED ISSUE #5132: Improve hint when provided array for destructuring has fewer elements than expected
### Description

```typ
#let (a, b, c) = (1, 2)
```

Gives:

```typ
error: not enough elements to destructure
  ┌─ 2024-10-05T17:11:22Z.typ:1:12
  │
1 │ #let (a, b, c) = (1, 2)
  │             ^
  │
  = hint: the provided array has a length of 2
```

```typ
#let (a, b, c) = (1, 2, 3, 4)
```

Gives:

```typ
error: too many elements to destructure
  ┌─ 2024-10-05T17:11:22Z.typ:1:5
  │
1 │ #let (a, b, c) = (1, 2, 3, 4)
  │      ^^^^^^^^^
  │
  = hint: the provided array has a length of 4, but the pattern expects 3 elements
```

I think the hints should mirror each other, therefore the first one should say:

```typ
  = hint: the provided array has a length of 2, but the pattern expects 3 elements
```

### Use Case

More useful/descriptive hints improves UX.

### PR DESCRIPTION
**Pull request by Ang149**
### Summary
This PR improves the error hint for destructuring assignment when the provided array has fewer elements than expected.

### Changes Made
- Updated the error message to provide clearer guidance on how to correct the destructuring assignment.
-  Modified existing test cases to cover new error hints and ensure they accurately reflect the changes made.

### Related Issue
Fixes #5132

### CODE REVIEW

File: tests/suite/scripting/destructuring.typ [hunk 1]
```diff
@@ -129,12 +129,12 @@
...
 --- destructuring-let-array-too-few-elements ---
 // Error: 13-14 not enough elements to destructure
-// Hint: 13-14 the provided array has a length of 2
+// Hint: 13-14 the provided array has a length of 2, but the pattern expects 3 elements
 #let (a, b, c) = (1, 2)
 
 --- destructuring-let-array-too-few-elements-with-sink ---
 // Error: 7-10 not enough elements to destructure
-// Hint: 7-10 the provided array has a length of 2
+// Hint: 7-10 the provided array has a length of 2, but the pattern expects 4 elements
```
Comment (PgBiel) [inline tests/suite/scripting/destructuring.typ:137]: This test shows that the number used is incorrect: `#let (..a, b, c, d) = (1, 2, 3)` works, and `a` is assigned `()` (empty array). Spreads shouldn't count in the expected number of elements.

**Review by PgBiel — COMMENTED** [review]
Nice!

File: tests/suite/scripting/destructuring.typ (hunk 1 above)
Comment (EpicEricEE) [reply tests/suite/scripting/destructuring.typ:137]: maybe with spreads, we could then even do "expects _at least_ 3 elements"

**Comment by laurmaedje** [comment]
Thanks!

**Pull request merged**